subsys/bootloader/provision/provision_flash.c

```diff
@@ -45,9 +55,16 @@ int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)
 		return -EINVAL;
 	}
 
-	for (size_t i = 0; i < CONFIG_SB_PUBLIC_KEY_HASH_LEN; i++) {
-		p_buf[i] = p_provision_data->pkd[(key_idx *
-				CONFIG_SB_PUBLIC_KEY_HASH_LEN) + i];
+	p_key = (u32_t *)&p_provision_data->pkd[key_idx *
```

Comment:
I'd prefer a few parentheses here to make it clearer what is being casted.

Reply:
Well, there is only one expression really.
`pointer->field[index]` 

There is no evaluation order taking place, so I'm not sure where to put the parentheses.

Reply:
Expression 1 (accessing member):
```
pointer->field
```
Expression 2 (accessing array element):
```
pointer->field[index]
```
Expression 3 (getting a pointer to the member):
```
&pointer->field[index]
```
Expression 4 (casting):
```
(type *)&pointer->field[index]
```

And now if you someone (not me) may ask - is this a pointer to the field or a pointer to the array element... or maybe a pointer to the pointer itself? What is going on here?

Reply:
That is true, my statement was imprecise. My point was that these types of accesses are quite common and rarely leaves people scratching their heads about operational precedence.

However, IMO having `(type *)&pointer->field[index]` is something which is typically done without using parentheses, and adding them here would reduce the readability. 

I can instead extract some helper variables to make it more clear.

Reply:
Well.... adding one here:
```
(type *)&(pointer->field[index])
```
should improve readability actually, but I do not have a strong opinion here. Just wanted to point out that expression here is a little more complex than you have stated above.

Reply:
I'm with rakons on this one just to make it a bit clearer.

Reply:
Will add this then :)